src/Renci.SshNet/Security/Cryptography/EcdsaKey.cs

```diff
@@ -207,34 +206,12 @@ public EcdsaKey(SshKeyData publicKeyData)
         /// <summary>
         /// Initializes a new instance of the <see cref="EcdsaKey"/> class.
         /// </summary>
-        /// <param name="curve">The curve name.</param>
+        /// <param name="curve">The curve oid.</param>
```

Comment:
That's a nasty break. Let's try to avoid it

Reply:
By reading the DER data, we get the curve oid string directly. It looks strange if we convert curve oid string to curve name  in PrivateKeyFile and then convert name back to oid string in Ecdsakey.

Reply:
Yeah, it's strange and unfortunate, but it's the publicly exposed contract, so my other comment applies here too. If you like, you can change it to accept either the name *or* oid

Reply:
We need to either document the breaking change in the release notes, or keep the current behavior but increase the code complexity (accept **either name or oid**). 😢

Reply:
Sure from coding perspective, it is not complex. It is just the method signature **definition**. Now the first parameter could be either name or oid. Would it break single responsibility principle?